### ISSUE TITLE
Behavior change for multiple mappings to same target after upgrade to 4.0

### ISSUE THREAD

**Issue by Nefarion** [riok/mapperly#1551]
**Describe the bug**
After the upgrade to 4.0, mapping multiple source properties to one target property throws `Error RMG074 : Multiple mappings are configured for the same target member <>`.  
Version 3.6.0 handled it fine.
What is the official workaround for this?
It is not mentioned in the breaking changes and there is no migration information.

**Declaration code**
```csharp
class Dto
{
    public object Article {get;set;}
}
class Obj
{
    public object Article {get;set;}
    public object ArticleNew {get;set;}
}
[Mapper(AllowNullPropertyAssignment = false, RequiredMappingStrategy = RequiredMappingStrategy.None)]
internal static partial class DtoMapper
{
    [MapProperty(nameof(Obj.Article), nameof(Dto.Article))]
    [MapProperty(nameof(Obj.ArticleNew), nameof(Dto.Article))]
    static partial void ObjToDto(Obj obj, Dto dto);
}
```
**Expected relevant generated code (Mapperly 3.6)**
Note: `ToDto` is the mapper function in DtoMapper for the relevant property type that does the mapping to another Dto type.
```csharp
[global::System.CodeDom.Compiler.GeneratedCode("Riok.Mapperly", "3.6.0.0")]
static partial void ObjToDto(global::Obj obj, global::Dto dto)
{
    if (obj.Article != null)
    {
        dto.Article = ToDto(obj.Article);
    }
    if (obj.ArticleNew != null)
    {
        dto.Article = ToDto(obj.ArticleNew);
    }
}
```

**Environment (please complete the following information):**
- Mapperly Version: 4.0
- Nullable reference types: enabled
- .NET Version: 8.0.400
- Target Framework: net8.0
- Compiler Version: 4.11.0-3.24365.8 (9e9c7c1d)
- C# Language Version: 12.0
- IDE: Rider 2024.3 EAP 4
- OS: Win22H2

**Comment by latonz** [comment]
Thanks for reporting this! I wasn’t even aware this was possible. It seems like this behavior would only make sense in combination with `AllowNullPropertyAssignment = false`. I’ll look into it further to see if it makes sense to enable `RMG074` specifically for `MapValue` and not for `MapProperty`.

In your example, you mentioned that NRT are enabled and that `Obj.Article` and `Obj.ArticleNew` are **not nullable**. Can you confirm if that’s actually the case, or are these members nullable? If they are indeed non-nullable, the null-check wouldn't seem necessary at all.

**Comment by latonz** [comment]
Should be fixed with `[4.1.0-next.2](https://www.nuget.org/packages/Riok.Mapperly/4.1.0-next.2)`. Let me know if it works for you…
